lib/bugsnag/middleware/delayed_job.rb

```diff
@@ -0,0 +1,73 @@
+module Bugsnag::Middleware
+  ##
+  # Attaches delayed_job information to an error report
+  class DelayedJob
+    def initialize(bugsnag)
+      @bugsnag = bugsnag
+    end
+
+    def call(report)
+      job = report.request_data[:delayed_job]
+      if job
+        job_data = {
+          :class => job.class.name,
+          :id => job.id
+        }
+        job_data[:priority] = job.priority if job.respond_to?(:priority)
+        job_data[:run_at] = job.run_at if job.respond_to?(:run_at)
+        job_data[:locked_at] = job.locked_at if job.respond_to?(:locked_at)
+        job_data[:locked_by] = job.locked_by if job.respond_to?(:locked_by)
+        job_data[:created_at] = job.created_at if job.respond_to?(:created_at)
+        job_data[:queue] = job.queue if job.respond_to?(:queue)
+
+        if job.respond_to?(:payload_object)
+          job_data[:active_job] = job.payload_object.job_data if job.payload_object.respond_to?(:job_data)
+          payload_data = construct_job_payload(job.payload_object)
+          report.context = payload_data[:method_name] if payload_data.include?(:method_name)
+          job_data[:payload] = payload_data
+        end
+
+        if job.respond_to?(:attempts)
+          max_attempts = (job.respond_to?(:max_attempts) && job.max_attempts) || Delayed::Worker.max_attempts
+          job_data[:attempts] = "#{job.attempts + 1} / #{max_attempts}"
```

Comment:
Would it make more sense to have 2 separate entries for this (an `attempt` and `max_attempts`)? (so the data is easier to deal with, and in case someone wants a custom filter/index on one of these for some reason)